### DISCUSSION TITLE
Multiple sequences

### DISCUSSION THREAD

**Discussion by mgpjm** [coldtype/coldtype#91] [Q&A]
Hi, I tried to find some information about this topic around but I didn't find anything. Let me know if I'm missing something out.

My goal with ColdType is to performe live coding performance, mainly in music context.
What I wish to reach is have a unique ColdType page with different sequences already written on it. Then call different animations with a midi controller.
As you can understand I'm looking for [something different then have a timeline](https://github.com/goodhertz/coldtype/discussions/75): it would be ideal have a text, and a way to call different kind of animations to apply on that, with the possibility to modify the current animation with midi controller.
Another cool implementation would be to manually trigger a specific animation, and let the animation be moduled by the live music instead of a static midi file or a pre-registred wav/mp3 file (like I saw on a some tutorials here).

Is that technically possible? There are some reference around?
Thank you very much, I really appreciate ColdType even if coding is not really my cup of tea.

m

**Comment by stenson** [comment]
Interesting question! Unfortunately I have to admit Coldtype is more tailored to a "prerecorded" filmmaking/editing mindset (closer to something like After Effects or a video editor) than to a live mindset, both because there are some tradeoffs between making things easier to edit or easier to do live, but also because I don't have much experience with live visuals. (Integrating code with live music is also unfortunately quite difficult outside of a DAW.)

Have you tried Processing/[p5.js](https://p5js.org/) at all? Or if you are looking to avoid coding altogether, have you experimented with something like [Videosync](https://videosync.showsync.com/)?

**Reply by mgpjm** [reply]
Thank for your answer. Yeah I understand your point, and for sure I will give a chances to processing/p5/drawbot. The fact is that I enjoy Coldtype because is really focused on typography and motions, which are exactly the directions where I'm interested. I didn't know Videosync, seems interesting! Also TocuhDesigner could be a direction.

BUT
I'm still wondering if I could use Coldtype as a launcher of animations: I could pre-code different animations and then call them by some midi input or something similar? Something like
[text]
[animation 1]
[animation 2]
[animation 3]
and apply them on my text  without having different files to open?

Yeah maybe I'm just using the wrong tool, but still I'm seeing some potential there for my scope :)
Thanks!

**Comment by mgpjm** [comment]
I try to put the question in a different way:
is it possible to call different pieces of code with a MIDI input?
So, instead of having different animations called by, for example, the AsciiTimeline, I would have them called by an input, to alternate, for example, different words on the screen.

Thank you!
m

**Reply by stenson** [reply]
Definitely! You can interpret the 0-1 midi values as booleans by doing something > 0.5, i.e. moving a slider past the midpoint will show an `A` and below the midpoint will show a `B`:

```python
from coldtype import *
from coldtype.midi.controllers import LaunchControlXL

@animation((1080, 1080), bg=1, rstate=1)
def render(f, rstate):
    nxl = LaunchControlXL(rstate.midi)
    toggle_state = nxl(10, 0) > 0.5
    
    if toggle_state:
        return (StSt("A", Font.MutatorSans(), 500)
            .align(f.a.r))
    else:
        return (StSt("B", Font.MutatorSans(), 300)
            .align(f.a.r))
```
